docs/content/docs/recipes/signals.md

```diff
@@ -0,0 +1,328 @@
+
+We'll begin by looking at the constructor arguments. `MouseTracker` takes in the `audience` and the `Signaler` instance from `initialObjects`:
+
+```typescript
+constructor(
+  public readonly audience: IServiceAudience<IMember>,
+  private readonly signaler: Signaler,
+) {
+  super();
+
+  /*...*/
+}
+```
+Whenever there is any change to a client's mouse position, we'll fire a `mousePositionChanged` event to notify listeners that updated data is available.
+
+```typescript
+export interface IMouseTrackerEvents extends IEvent {
+    (event: "mousePositionChanged", listener: () => void): void;
+}
+```
+When a member leaves the audience (e.g., a client disconnects from the container), we would want to remove their presence from our local data. After this is done, we would have to let the view know that the local presence data has changed. To do this, we emit our `mousePositionChanged` event so the view knows to re-render:
+
+```typescript
+this.audience.on("memberRemoved", (clientId: string, member: IMember) => {
+  const clientIdMap = this.posMap.get(member.userId);
+  if (clientIdMap !== undefined) {
+      clientIdMap.delete(clientId);
+
+      //If UserID has no connected clients, remove user from local data
+      if (clientIdMap.size === 0) {
+          this.posMap.delete(member.userId);
+      }
+  }
+  this.emit("mousePositionChanged");
+});
+```
+`MouseTracker` must share information to all connected clients about where the local client's mouse position is. This is where `Signaler` comes in handy, as it specializes in communicating transient data to connected clients within a Fluid application.
+
+To track the client's mouse position, we use the `mousemove` event to obtain the client's x and y coordinates. To alert the connected clients of a mouse position change, we then submit a `mouseSignalType` signal with the client's userID and updated position value as the payload:
+
+```typescript
+window.addEventListener("mousemove", (e) => {
+    const position: IMousePosition = {
+        x: e.clientX,
+        y: e.clientY,
+    };
+    this.sendMouseSignal(position);
+});
+```
+```typescript
+/**
+* Alert all connected clients that there has been a change to a client's mouse position
+*/
+private sendMouseSignal(position: IMousePosition) {
+    this.signaler.submitSignal(
+        MouseTracker.mouseSignalType,
+        { userId: this.audience.getMyself()?.userId, pos: position },
+    );
+}
+```
+But what is the point of sending a signal if nobody is listening to it? To listen to the `mouseSignalType`, we use `Signaler`'s `onSignal` function. We then update the local data using the payload information from the signal. To display the new presence data, we then emit `mousePositionChanged` to let the view know to re-render:
+
+```typescript
+this.signaler.onSignal(MouseTracker.mouseSignalType, (clientId: string, local: boolean, payload: Jsonable) => {
```

Comment:
Also, for what it is worth, the example app that was linked to in the first paragraph of the article doesn't have these type specifications in the MouseTracker.ts file. For example, this line is like this in the example app: 

```
this.signaler.onSignal(MouseTracker.mouseSignalType, (clientId, local, payload) => {
```